OIDC "Permission must not be empty" with imported test realm since 3.5.0

### Describe the bug

In certain projects, we use a realm.json to initialize the keycloak dev instance for our tests.
This did work fine prior 3.5.0

Since 3.5.0 we do get an exception as soon as we try to send an authorized request to a REST api during a test:
```
Caused by: java.lang.IllegalArgumentException: Permission name must not be empty
                at io.quarkus.security.StringPermission.validateAndTrim(StringPermission.java:48)
                at io.quarkus.security.StringPermission.<init>(StringPermission.java:26)
                at io.quarkus.oidc.runtime.OidcUtils.transformScopesToPermissions(OidcUtils.java:314)
                at io.quarkus.oidc.runtime.OidcUtils$1.<init>(OidcUtils.java:288)
                at io.quarkus.oidc.runtime.OidcUtils.addTokenScopesAsPermissions(OidcUtils.java:286)
                at io.quarkus.oidc.runtime.OidcUtils.setSecurityIdentityPermissions(OidcUtils.java:281)
                at io.quarkus.oidc.runtime.OidcUtils.validateAndCreateIdentity(OidcUtils.java:270)
                at io.quarkus.oidc.runtime.OidcIdentityProvider.createSecurityIdentityWithOidcServer(OidcIdentityProvider.java:284)
```

As nothing is noted as a breaking change concerning oidc in 3.5.0 and the imported realm.json seems valid, I must assume that this is bug?

PS: Mabey our keycloak "test tool" to add users, roles etc. could be distilled and integrated in quarkus?

### How to Reproduce?

Run tests:
https://github.com/HerrDerb/quarkus-issue/tree/oidc-3-5-0-issue

Switch to Quarkus version 3.4.3 and the test will succeed.

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java
@@ -208,6 +208,9 @@ public final class OidcUtils {
             return convertJsonArrayToList((JsonArray) claimValue);
         } else if (claimValue != null) {
             String sep = rolesConfig.getRoleClaimSeparator().isPresent() ? rolesConfig.getRoleClaimSeparator().get() : " ";
+            if (claimValue.toString().isBlank()) {
+                return Collections.emptyList();
+            }
             return Arrays.asList(claimValue.toString().split(sep));
         } else {
             return Collections.emptyList();
@@ -237,6 +240,10 @@ public final class OidcUtils {
     private static List<String> convertJsonArrayToList(JsonArray claimValue) {
         List<String> list = new ArrayList<>(claimValue.size());
         for (int i = 0; i < claimValue.size(); i++) {
+            String claimValueStr = claimValue.getString(i);
+            if (claimValueStr == null || claimValueStr.isBlank()) {
+                continue;
+            }
             list.add(claimValue.getString(i));
         }
         return list;

--- a/integration-tests/oidc-tenancy/src/main/java/io/quarkus/it/keycloak/OidcResource.java
+++ b/integration-tests/oidc-tenancy/src/main/java/io/quarkus/it/keycloak/OidcResource.java
@@ -238,6 +238,16 @@ public class OidcResource {
     }
 
     @POST
+    @Path("accesstoken-empty-scope")
+    @Produces("application/json")
+    public String testAccessTokenWithEmptyScope(@QueryParam("kid") String kid, @QueryParam("sub") String subject) {
+        return "{\"access_token\": \"" + jwt(null, subject, kid, true) + "\"," +
+                "   \"token_type\": \"Bearer\"," +
+                "   \"refresh_token\": \"123456789\"," +
+                "   \"expires_in\": 300 }";
+    }
+
+    @POST
     @Path("opaque-token")
     @Produces("application/json")
     public String testOpaqueToken(@QueryParam("kid") String kid) {
@@ -290,6 +300,10 @@ public class OidcResource {
     }
 
     private String jwt(String audience, String subject, String kid) {
+        return jwt(audience, subject, kid, false);
+    }
+
+    private String jwt(String audience, String subject, String kid, boolean withEmptyScope) {
         JwtClaimsBuilder builder = Jwt.claim("typ", "Bearer")
                 .upn("alice")
                 .preferredUserName("alice")
@@ -300,6 +314,10 @@ public class OidcResource {
         }
         if (subject != null) {
             builder.subject(subject);
+        }
+
+        if (withEmptyScope) {
+            builder.claim("scope", "");
         }
 
         return builder.jws().keyId(kid)

--- a/integration-tests/oidc-tenancy/src/test/java/io/quarkus/it/keycloak/BearerTokenAuthorizationTest.java
+++ b/integration-tests/oidc-tenancy/src/test/java/io/quarkus/it/keycloak/BearerTokenAuthorizationTest.java
@@ -517,15 +517,24 @@ public class BearerTokenAuthorizationTest {
                                     + "introspection_client_id:none,introspection_client_secret:none,active:true,userinfo:alice,cache-size:0"));
         }
 
+        // verifies empty scope claim makes no difference (e.g. doesn't cause NPE)
+        RestAssured.given().auth().oauth2(getAccessTokenWithEmptyScopeFromSimpleOidc("2"))
+                .when().get("/tenant/tenant-oidc-introspection-only/api/user")
+                .then()
+                .statusCode(200)
+                .body(equalTo(
+                        "tenant-oidc-introspection-only:alice,client_id:client-introspection-only,"
+                                + "introspection_client_id:none,introspection_client_secret:none,active:true,userinfo:alice,cache-size:0"));
+
         RestAssured.given().auth().oauth2(getAccessTokenFromSimpleOidc("987654321", "2"))
                 .when().get("/tenant/tenant-oidc-introspection-only/api/user")
                 .then()
                 .statusCode(401);
 
         RestAssured.when().get("/oidc/jwk-endpoint-call-count").then().body(equalTo("0"));
-        RestAssured.when().get("/oidc/introspection-endpoint-call-count").then().body(equalTo("4"));
+        RestAssured.when().get("/oidc/introspection-endpoint-call-count").then().body(equalTo("5"));
         RestAssured.when().post("/oidc/disable-introspection").then().body(equalTo("false"));
-        RestAssured.when().get("/oidc/userinfo-endpoint-call-count").then().body(equalTo("4"));
+        RestAssured.when().get("/oidc/userinfo-endpoint-call-count").then().body(equalTo("5"));
         RestAssured.when().get("/cache/size").then().body(equalTo("0"));
     }
 
@@ -694,13 +703,21 @@ public class BearerTokenAuthorizationTest {
     }
 
     private String getAccessTokenFromSimpleOidc(String subject, String kid) {
+        return getAccessTokenFromSimpleOidc(subject, kid, "/oidc/accesstoken");
+    }
+
+    private String getAccessTokenWithEmptyScopeFromSimpleOidc(String kid) {
+        return getAccessTokenFromSimpleOidc("123456789", kid, "/oidc/accesstoken-empty-scope");
+    }
+
+    private static String getAccessTokenFromSimpleOidc(String subject, String kid, String tokenEndpoint) {
         String json = RestAssured
                 .given()
                 .queryParam("sub", subject)
                 .queryParam("kid", kid)
                 .formParam("grant_type", "authorization_code")
                 .when()
-                .post("/oidc/accesstoken")
+                .post(tokenEndpoint)
                 .body().asString();
         JsonObject object = new JsonObject(json);
         return object.getString("access_token");

